### PR TITLE
[main] Removing redundant `%clean` stages from the spec files.

### PR DESCRIPTION
**Pull request by PawelWMS**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [X] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [X] Any updated packages successfully build (or no packages were changed)
- [X] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [X] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

While working on a fix for `c-ares` test fix I've noticed that the RPM builds behave in an unexpected way when a stage preceding the `%clean` stage fails. In case of `c-ares` this resulted in the clean-up happening before the RPMs were created and caused a build error, since no files inside `%{buildroot}` could be found. Since `rpmbuild` automatically cleans up the build now without the explicit `%clean` stage and we don't even need the clean-up with the way we build packages, I'm removing all `%clean` sections from the specs files to avoid any other flaky builds.

**NOTE**: the PR contains identical changes inside 108 spec files. The only exception is `c-ares.spec` where I've also performed minor clean-up. The other modified files are:

- a new GitHub check making sure we don't add specs with a `%clean` stage anymore and
- 4 .txt manifests.

###### Change Log  

- Removed the `%clean` stage from all spec files.
- Added a GitHub check for the `%clean` stage.

###### Does this affect the toolchain?  

Yes.

###### Test Methodology

- Local package build for `c-ares`.

### CODE REVIEW

File: .github/workflows/check-clean-stage.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      # Checkout the branch of our repo that triggered this action
+      - name: Workflow trigger checkout
+        uses: actions/checkout@v2
+
+      - name: Get base commit for PRs
+        if: ${{ github.event_name == 'pull_request' }}
+        run: |
+          git fetch origin ${{ github.base_ref }}
+          echo "base_sha=$(git rev-parse origin/${{ github.base_ref }})" >> $GITHUB_ENV 
+          echo "Merging ${{ github.sha }} into ${{ github.base_ref }}"
```
Comment (mfrw) [inline .github/workflows/check-clean-stage.yml:24]: I do not see an actual merge happening. Am I missing anything trivial ?

File: .github/workflows/check-clean-stage.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+      - name: Get base commit for Pushes
+        if: ${{ github.event_name == 'push' }}
+        run: |
+          git fetch origin ${{ github.event.before }}
+          echo "base_sha=${{ github.event.before }}" >> $GITHUB_ENV 
+          echo "Merging ${{ github.sha }} into ${{ github.event.before }}"
```
Comment (mfrw) [inline .github/workflows/check-clean-stage.yml:31]: Same comment as above^

File: SPECS/ltrace/ltrace.spec [hunk 3]
```diff
@@ -42,16 +42,16 @@ make install DESTDIR=%{buildroot}
...
-%clean
-rm -rf %{buildroot}/*
-
 %files
 %defattr(-,root,root)
 %license COPYING
 %{_sysconfdir}/ltrace.conf
 %{_bindir}/*
 %{_datadir}
 %changelog
```
Comment (mfrw) [inline SPECS/ltrace/ltrace.spec:51]: Maybe we should add a new line while we are here:
```suggestion

%changelog
```

**Review by mfrw — APPROVED** [review]
Changes LGTM :)

Although there is 1 suggestion and 1 clarification requested :)

File: SPECS/ltrace/ltrace.spec (hunk 3 above)
Comment (PawelWMS) [reply SPECS/ltrace/ltrace.spec:51]: Sure.

**Review by jslobodzian — APPROVED** [review]

**Pull request merged**